machine/models/seq2seq.py

```diff
@@ -32,3 +37,15 @@ def forward(self, inputs, input_lengths=None, targets={},
                                      function=self.decode_function,
                                      teacher_forcing_ratio=teacher_forcing_ratio)
         return result
+
+    def _init_weights(self, uniform_init=0.0, glorot_init=False):
+        # initialize weights using uniform distribution
+        if uniform_init > 0.0:
+            for p in self.parameters():
+                p.data.uniform_(-uniform_init, uniform_init)
+
+        # xavier/glorot initialization if glorot_init
+        if glorot_init:
```

Comment:
I don't know much about initialisations, so mostly a question: does the glorot initialisation depend on the probability of the uniform? If not they are both specified does the glorot_init overwrite the uniform_init paramater? In the tests it seems that the are not mutually exclusive. Perhaps we could add some docstring explaining how they behave/interact?

Reply:
Yes if both are passed, then in this case Glorot overrides the uniform initialization for all parameters except biases. You usually tend to use either uniform or glorot/xavier at a time so I could add a docstring about this but it's not super likely that someone activates them both at the same time. Also code-wise this is exactly how openNMT does it.

Reply:
Alright, makes sense. My confusion was mostly coming from my ignorance:).